File: MemoryManagement/MemManage.java
public class MemManage {
    public static void main(String[] args) {
        String s = new String("allocate memory"); // allocate memory in heap
        printString(s);
        // do not need to explicetly dellocate mem. GC will take care
    }

    public static void printString(String s) {
        System.out.println(s);
    }
}

File: MemoryManagement/mem_manage.rs
fn main() {
    let s = String::from("allocate memory"); // allocate memory in heap
    print_string(&s); // borrowing s and now the mem of string belongs to the function. s does not have anything
    // do not need to explicetly dellocate mem. Rust's ownership handles it
}

fn print_string(s: &String) {
    println!("{}", s);
}